commit a8657e68242772354aeaa0ccd0f35f1bcf1cceaa
Author: Keyon Jie <yang.jie@linux.intel.com>
Date:   Fri Nov 16 18:47:04 2018 -0600

    ASoC: acpi: fix: continue searching when machine is ignored
    
    [ Upstream commit a3e620f8422832afd832ad5e20aa97d0c72bada8 ]
    
    The machine_quirk may return NULL which means the acpi entries should be
    skipped and search for next matched entry is needed, here add return
    check here and continue for NULL case.
    
    Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>
    Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
    Signed-off-by: Mark Brown <broonie@kernel.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/sound/soc/soc-acpi.c b/sound/soc/soc-acpi.c
index b8e72b52db30..4fb29f0e561e 100644
--- a/sound/soc/soc-acpi.c
+++ b/sound/soc/soc-acpi.c
@@ -10,11 +10,17 @@ struct snd_soc_acpi_mach *
 snd_soc_acpi_find_machine(struct snd_soc_acpi_mach *machines)
 {
 	struct snd_soc_acpi_mach *mach;
+	struct snd_soc_acpi_mach *mach_alt;
 
 	for (mach = machines; mach->id[0]; mach++) {
 		if (acpi_dev_present(mach->id, NULL, -1)) {
-			if (mach->machine_quirk)
-				mach = mach->machine_quirk(mach);
+			if (mach->machine_quirk) {
+				mach_alt = mach->machine_quirk(mach);
+				if (!mach_alt)
+					continue; /* not full match, ignore */
+				mach = mach_alt;
+			}
+
 			return mach;
 		}
 	}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1063_a8657e68242772354aeaa0ccd0f35f1bcf1cceaa.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1063_a8657e68242772354aeaa0ccd0f35f1bcf1cceaa.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
